packages/formstr-app/src/containers/CreateFormNew/components/FormDetails/index.tsx

```diff
@@ -84,53 +84,99 @@ export const FormDetails: React.FC<FormDetailsProps> = ({
     viewKey?: string
   ) => {
     if (!userPub) return;
-    setSavedOnNostr("saving");
-    let existingListFilter = {
-      kinds: [KINDS.myFormsList],
-      authors: [userPub],
-    };
-    let pool = new SimplePool();
-    let existingList = await pool.querySync(
-      getDefaultRelays(),
-      existingListFilter
-    );
-    let forms: Tag[] = [];
-    if (existingList[0]) {
-      let formsString = await window.nostr.nip44.decrypt(
-        userPub,
-        existingList[0].content
-      );
-      try {
-        forms = JSON.parse(formsString);
-      } catch (e) {
-        console.error("My Forms List is not parseable");
-        return;
-      }
-    }
-    let key = `${formAuthorPub}:${formId}`;
-    if (forms.map((f) => f[1]).includes(key)) {
-      setSavedOnNostr("saved");
+    
+    if (saveToMyForms.isRunning) {
+      console.log('Save operation already in progress');
       return;
     }
-    let secrets = `${formAuthorSecret}`;
-    if (viewKey) secrets = `${secrets}:${viewKey}`;
-    forms.push(["f", `${key}`, `${relay}`, `${secrets}`]);
-    let encryptedString = await window.nostr.nip44.encrypt(
-      userPub,
-      JSON.stringify(forms)
-    );
-    let myFormEvent: UnsignedEvent = {
-      kind: KINDS.myFormsList,
-      content: encryptedString,
-      pubkey: userPub,
-      tags: [],
-      created_at: Math.round(Date.now() / 1000),
-    };
-    let signedEvent = await window.nostr.signEvent(myFormEvent);
-    await Promise.allSettled(pool.publish(getDefaultRelays(), signedEvent));
-    setSavedOnNostr("saved");
+    
+    saveToMyForms.isRunning = true;
+    setSavedOnNostr("saving");
```

Comment:
why are we using  saveToMyForms.isRunning  and setSavedOnNostr to check if form is saved